department-of-veterans-affairs/caseflow
# APPEALS-9098

Resolves [#9098](https://vajira.max.gov/browse/APPEALS-9098)

### Description
Adds a "Return to Board Intake" task action for VHA CAMO


### Acceptance Criteria
- [x] The following occurs when the VHA CAMO user selects the "Return to Board Intake" task action
  - [x] A modal appears to confirm intent
  - [x] Modal Header: Return to Board Intake
  - [x] Modal information text: This appeal will be sent to Board Intake. Why is this appeal being returned?
  - [x] User is required to select why the appeal is being returned from the following options
    1. Duplicate
    2. HLR Pending
    3. SC Pending
    4. Not VHA related
    5. Clarification needed from appellant
    6. No VHA decision
    7. Other
      - [x] When 'Other' is selected, user is required to provide additional context via additional text entry box as shown in designs
      - [x] Field type: required free text field
   - [x] User may add additional details for this action
     - [x] Optional free text field
   - [x] Modal contains the following selectable options
     - [x] Cancel - User returns to case details screen and selection is not retained
  - [x] Return
       - [x] Select to execute task action
       - [x] Option disabled until required field(s) populated
- [x] The following occurs when the Return to Board Intake task action is executed
  - [x]   The appeal appears on the Completed tab of the VHA CAMO queue
  - [x]   The appeal appears on the Ready for Review tab of the BVA Intake organization queue
  - [x]   User is returned to the VHA CAMO organization queue's completed tab
  - [x]   A message appears confirming the case has been returned to Board Intake

### Testing Plan
1. To generate test data
    1.  Open a rails console (`rails c`)
    2. Generate some cases: `10.times.map { FactoryBot.create(:vha_document_search_task) }`
    3.  Make sure they're properly assigned: `PreDocketTask.all.each { |t| t.update!(assigned_to: BvaIntake.singleton) }`
4.  Log in as CAMOADMIN
6. Navigate to the VHA CAMO queue: http://localhost:3000/organizations/vha-camo
7. Click on a case
8. You should see an actions dropdown with 'Return to Board Intake'. Click it.
<img width="1288" alt="image" src="https://user-images.githubusercontent.com/17840614/219151307-bac5860e-2a07-404a-9050-5c2e89bec1f3.png">

9. A Return to Board Intake modal should appear. The return button should be disabled.
10. Select 'Other' for the appeal reason
11. Add a reason and context. The return button should now be enabled.

<img width="474" alt="image" src="https://user-images.githubusercontent.com/17840614/219152991-a368fe99-a516-4c3e-979e-6252588abcc6.png">

13. Click 'Return'
14. You should see this screen
<img width="1257" alt="image" src="https://user-images.githubusercontent.com/17840614/219153443-04149287-7dca-4196-8731-572cc7555fb5.png">

15. Switch users to BVAISHAW
16. Go to the organizational queue for BVA (http://localhost:3000/organizations/bva-intake)
17. Go to the Ready for Review tab. The task just returned to board should be there. Click it.
18. Scroll down to Case Timeline and click the View Task Instructions button.
19. You should see the task instructions you entered
<img width="864" alt="image" src="https://user-images.githubusercontent.com/17840614/219154660-dcc649eb-c19f-4910-a209-97705e146733.png">





### User Facing Changes
BEFORE
---
<img width="1244" alt="image" src="https://user-images.githubusercontent.com/17840614/219159843-7d701566-b1cb-40f0-b99a-e4d6e8748488.png">


AFTER
---
<img width="1330" alt="image" src="https://user-images.githubusercontent.com/17840614/219159145-ceeb84e6-8758-4c2c-96ef-c9e76108bfb3.png">
<img width="483" alt="image" src="https://user-images.githubusercontent.com/17840614/219159216-601384ea-4ed5-4b7b-9bf6-b79a1d1e2b46.png">
<img width="483" alt="image" src="https://user-images.githubusercontent.com/17840614/219159290-216d8a7e-c73e-400b-bdea-e03f36495eca.png">
<img width="496" alt="image" src="https://user-images.githubusercontent.com/17840614/219159382-e7d794f0-38f0-4bfc-97ee-5c26a9b5822a.png">
